modules/broker-core/src/main/java/io/ballerina/messaging/broker/core/store/dao/impl/MessageDaoImpl.java

```diff
@@ -37,15 +40,18 @@
 
     private final DtxCrudOperationsDao dtxCrudOperationsDao;
 
+    private final Map<Xid, Long> xidToInternalIdMap;
```

Comment:
1. We might have to repopulate this when the server is restarted.
2. Don't we remove items from this map? Looks like the items are accumulating.